icu4c/source/common/locid.cpp

```diff
@@ -590,6 +590,12 @@ Locale::clone() const {
 bool
 Locale::operator==( const   Locale& other) const
 {
+    if (other.fullName == fullName) {
```

Comment:
This PR needs to be rebased on current HEAD, class Locale no longer has this `fullName` member.